File: src/main/java/ui_adapter/model/TestStep.java
package ui_adapter.model;

/**
 * Represents a single test step in a test case.
 * Contains the step description, test data, and expected result.
 */
public class TestStep {
    private String stepSummary;
    private String testData;
    private String expectedResult;

    public TestStep() {
    }

    public TestStep(String stepSummary, String testData, String expectedResult) {
        this.stepSummary = stepSummary;
        this.testData = testData;
        this.expectedResult = expectedResult;
    }

    public String getStepSummary() {
        return stepSummary;
    }

    public void setStepSummary(String stepSummary) {
        this.stepSummary = stepSummary;
    }

    public String getTestData() {
        return testData;
    }

    public void setTestData(String testData) {
        this.testData = testData;
    }

    public String getExpectedResult() {
        return expectedResult;
    }

    public void setExpectedResult(String expectedResult) {
        this.expectedResult = expectedResult;
    }

    @Override
    public String toString() {
        StringBuilder sb = new StringBuilder();
        if (stepSummary != null && !stepSummary.isEmpty()) {
            sb.append("Step: ").append(stepSummary).append("\n");
        }
        if (testData != null && !testData.isEmpty()) {
            sb.append("Test Data: ").append(testData).append("\n");
        }
        if (expectedResult != null && !expectedResult.isEmpty()) {
            sb.append("Expected Result: ").append(expectedResult).append("\n");
        }
        return sb.toString();
    }
}

File: python-code-base/playwright_ui_adapter/model/test_step.py
"""Test step model for individual steps within a test case."""

from dataclasses import dataclass
from typing import Optional


@dataclass
class TestStep:
    """
    Represents a single step within a test case.
    
    Test steps are loaded from Excel files and define the actions
    to perform and expected results for validation.
    
    Attributes:
        step_summary: The action to perform (e.g., "Login with valid credentials")
        test_data: Optional test data for the step (e.g., "username: admin, password: admin123")
        expected_result: Expected outcome for validation (e.g., "User is logged in successfully")
    """
    
    step_summary: str
    test_data: Optional[str] = None
    expected_result: Optional[str] = None
    
    def __str__(self) -> str:
        parts = [f"Action: {self.step_summary}"]
        if self.test_data:
            parts.append(f"Data: {self.test_data}")
        if self.expected_result:
            parts.append(f"Expected: {self.expected_result}")
        return ", ".join(parts)
    
    def __repr__(self) -> str:
        return f"TestStep({self.step_summary})"
